src/d3a/models/area/event_dispatcher.py

```diff
@@ -108,13 +109,14 @@ def _should_dispatch_to_strategies(self, event_type):
             return self.area.events.is_connected and self.area.events.is_enabled
 
     def event_listener(self, event_type: Union[MarketEvent, AreaEvent], **kwargs):
+        current_tick = kwargs.pop('current_tick', None)
         if event_type is AreaEvent.TICK and \
                 self._should_dispatch_to_strategies(event_type):
             self.area.tick_and_dispatch()
         if event_type is AreaEvent.MARKET_CYCLE:
             self.area.cycle_markets(_trigger_event=True)
         elif event_type is AreaEvent.ACTIVATE:
-            self.area.activate()
+            self.area.activate(current_tick=current_tick)
```

Comment:
I think you can pass kwargs here instead of the current_tick. Assuming that it works, it will make it more future-proof because it will work if we ever need to add more arguments to the broadcast_activate method.

Reply:
https://github.com/gridsingularity/d3a/pull/1021#discussion_r562698227